### PR TITLE
re-create addon client cert once bootstrap kubeconfig changes

### PR DESCRIPTION
**Pull request by elgnay**
Signed-off-by: Yang Le <yangle@redhat.com>

### CODE REVIEW

File: pkg/clientcert/cert_controller.go [hunk 1]
```diff
@@ -187,7 +191,12 @@ func (c *clientCertificateController) sync(ctx context.Context, syncCtx factory.
 	// create a csr to request new client certificate if
 	// a. there is no valid client certificate issued for the current cluster/agent
 	// b. client certificate exists and has less than a random percentage range from 20% to 25% of its life remaining
-	if c.hasValidClientCertificate(secret) {
+
+	hasValidClientCert := c.hasValidClientCertificate(secret)
+	switch {
+	case hasValidClientCert && c.AdditonalSecretDataSensitive && !c.hasAdditonalSecretData(secret):
```
Comment (qiujian16) [inline pkg/clientcert/cert_controller.go]: maybe

```go
switch{
case !hasValidClientCert:
    // record event and break
case c.AdditonalSecretDataSensitive && !c.hasAdditonalSecretData(secret):
   // record event and break
default:
  ...
}
```

Comment (elgnay) [reply pkg/clientcert/cert_controller.go]: Updated.

**Pull request merged**